latch/types/file.py

```diff
@@ -78,7 +83,27 @@ def downloader():
                     # todo(kenny) is this necessary?
                     and ctx.inspect_objects_only is False
                 ):
-                    self.path = ctx.file_access.get_random_local_path(self._remote_path)
+                    local_path_hint = self._remote_path
+                    if is_absolute_node_path.match(self._remote_path) is not None:
+                        data = execute(
+                            gql.gql(
+                                """
+                            query nodeIdQ($argPath: String!) {
+                                ldataResolvePathData(
+                                    path: $argPath
+                                ) {
+                                    name
+                                }
+                            }
+                            """
+                            ),
+                            {"argPath": self._remote_path},
+                        )["ldataResolvePathData"]
+
+                        if data["name"] is not None:
```

Comment:
```suggestion
                        if data is not None and data["name"] is not None:
```

`data` will be `None` if the node doesn't exist